x-pack/plugins/alerting/server/rules_client/tests/bulk_edit.test.ts

```diff
@@ -251,6 +253,7 @@ describe('bulkEdit()', () => {
             type: 'alert',
             attributes: expect.objectContaining({
               tags: ['foo', 'test-1'],
+              revision: 1,
```

Comment:
So this is the issue with a lot of our tests: while the test says `should add new tag` it _also_ verifies that the revision gets incremented. This has a few very important downsides that make it much harder to maintain them when there are a lot of them:

- It makes testing (for the revision update logic) implicit and buried under all the other testing logic.
- It makes tests more brittle and prone to false positive errors: when the revision update logic changes, it will make tests like `should add new tag` red although the logic of updating the tags themselves still works correctly.

I think this should be a separate test, and along with other tests, a whole suite of tests checking in what conditions and cases the revision gets incremented and in what cases it doesn't. 

Let's please extract tests for revision either into a separate test suite, or just have separate `it` blocks for them in every relevant test suite under `rules_client/tests`, and be very specific about what we check in them.